Pod/Classes/PINRemoteImageCaching.h

```diff
@@ -0,0 +1,45 @@
+//
+//  PINImageCacheProtocol.h
+//  Pods
+//
+//  Created by Aleksei Shevchenko on 7/25/16.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol PINRemoteImageCaching;
+typedef void (^PINRemoteImageCachingObjectBlock)(id<PINRemoteImageCaching> cache, NSString *key, id __nullable object);
+
+/**
+ *  Image Cache is responsible for actual image caching.
+ */
+@protocol PINRemoteImageCaching <NSObject>
+
+//******************************************************************************************************
+// Memory cache methods
+//******************************************************************************************************
+- (nullable id)objectCachedInMemoryForKey:(NSString *)key;
+- (void)cacheObjectInMemory:(id)object forKey:(NSString *)key withCost:(NSUInteger)cost;
+
+//******************************************************************************************************
+// Disk cache methods
+//******************************************************************************************************
+- (nullable id)objectCachedOnDiskForKey:(NSString *)key;
+- (void)objectCachedOnDiskForKey:(NSString *)key block:(nullable PINRemoteImageCachingObjectBlock)block;
+- (void)cacheObjectOnDisk:(id)object forKey:(NSString *)key;
+
+
+- (BOOL)hasObjectForKey:(NSString *)key;
```

Comment:
You're right. I was reading too much into the previous method declarations and lost track of what this method returns. New suggestion:

`- (BOOL)containsObjectWithKey:(NSString *)key;`